projects/RabbitMQ.Client/client/impl/SocketFrameHandler.cs

```diff
@@ -226,37 +225,32 @@ public void Close()
             {
                 if (!_closed)
                 {
-                    try
-                    {
-                        _channelWriter.Complete();
-                        _writerTask.GetAwaiter().GetResult();
-                        _pipeReader.Complete();
-                        _pipeWriter.Complete();
-                    }
-                    catch (Exception)
-                    {
-                    }
+                    _channelWriter.TryComplete();
+                    _frameWriter.TryComplete();
+                    try { _writerTask.GetAwaiter().GetResult(); } catch { }
 
-                    try
-                    {
-                        _socket.Close();
-                    }
-                    catch (Exception)
-                    {
-                        // ignore, we are closing anyway
-                    }
-                    finally
+                    if (_pipe != null)
                     {
-                        _closed = true;
+                        try { _pipe.Input?.CancelPendingRead(); } catch { }
+                        try { _pipe.Input?.Complete(); } catch { }
+                        try { _pipe.Output?.CancelPendingFlush(); } catch { }
+                        try { _pipe.Output?.Complete(); } catch { }
+                        try { using (_pipe as IDisposable) { } } catch { }
                     }
+
+                    try { _socket.Close(); } catch { }
+                    _closed = true;
                 }
             }
         }
 
         public void SendHeader()
         {
-            byte[] headerBytes = ArrayPool<byte>.Shared.Rent(8);
-            Encoding.ASCII.GetBytes("AMQP", 0, 4, headerBytes, 0);
+            Span<byte> headerBytes = stackalloc byte[8];
```

Comment:
why stackalloc + copy you could write directly into the pipe

Reply:
You are right :) will fix that.